packages/app/src/app/utils/Loadable.js

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import Loading from '../components/Loading';
+
+export default loader =>
+  class extends React.Component {
+    state = {
+      LoadedComponent: null,
+    };
+    componentDidMount() {
+      loader().then(module => {
+        this.setState({
+          LoadedComponent: module.default,
+        });
+      });
+    }
+    render() {
+      const { LoadedComponent } = this.state;
+
+      if (LoadedComponent) {
+        return <LoadedComponent />;
+      }
+
+      return <Loading />;
```

Comment:
Can we show the spinner after a set time, like 1s? That will make perceived loading speed faster and is what Loadable has been doing.

Reply:
Jup, for sure, let me fix :)